client/src/store/modules/projects/mutations.ts

```diff
@@ -1,13 +1,34 @@
-import {SET_CUSTOMERS, SET_PROJECTS} from './project-types';
+import {SET_CUSTOMERS, SET_PROJECTS, SET_LINES, SET_PROGRAMS, SET_DOMAINS, SET_TYPES} from './project-types';
 import {IProject} from '../../../shared/interfaces/IProject';
 import {IProjectState} from './index';
+import {ICustomer} from '../../../shared/interfaces/ICustomer';
+import {ILine} from '../../../shared/interfaces/ILine';
+import {IProgram} from '../../../shared/interfaces/IProgram';
+import {IDomain} from '../../../shared/interfaces/IDomain';
+import {IType} from '../../../shared/interfaces/IType';
 
 export const mutations = {
   [SET_PROJECTS](state: IProjectState, payload: IProject[]) {
     state.projects = payload;
     state.loading = false;
   },
-  [SET_CUSTOMERS](state: IProjectState, payload: String[]) {
+  [SET_LINES](state: IProjectState, payload: ILine[]) {
+    state.lines = payload;
+    state.loading = false;
```

Comment:
Extract loading to separate mutation and just commit after all SET_'s

Reply:
done